style/animation.rs

```diff
@@ -1015,11 +1015,55 @@ impl ElementAnimationSet {
             if transition.state == AnimationState::Finished {
                 continue;
             }
-            if transitioning_properties.contains(transition.property_animation.property_id()) {
+            // Step 3 in https://drafts.csswg.org/css-transitions/#starting
+            // "If the element has a running transition or completed transition for the property, 
+            // and there is not a matching transition-property value, then implementations 
+            // must cancel the running transition or remove the completed transition from the set of completed transitions."
+            if !transitioning_properties.contains(transition.property_animation.property_id()) {
+                transition.state = AnimationState::Canceled;
+                self.dirty = true;
                 continue;
             }
-            transition.state = AnimationState::Canceled;
-            self.dirty = true;
+            
+            let transition_property_id = transition.property_animation.property_id();
```

Comment:
Move this upwards so that you can use `if !transitioning_properties.contains(transition_property_id)`